torch/csrc/jit/interpreter.cpp

```diff
@@ -178,31 +453,28 @@ Operation createEvalOperation(CppOp * op) {
       output_edges.emplace_back(handle_in->forward_inputs, i);
     auto values = engine.execute(handle_in->forward_outputs, v_inputs, true, create_graph, output_edges);
     for(auto & v : values)
-      outputs.push_back(builder.addOutput(v));
-    builder.writeTo(outputs);
+      stack.push_back(builder.addOutput(v));
+    builder.writeTo(stack);
+    return 0;
   };
 }
 
-using tensor_list = std::vector<at::Tensor>;
-
-tensor_list unsafeToTensorListShare(const list_of_retainable & inputs) {
-  tensor_list tinputs;
-  tinputs.reserve(inputs.size());
-  for(auto & i : inputs) {
-    tinputs.push_back(unsafeToTensorShare(i));
-  }
-  return tinputs;
-}
-void moveToListOfRetainables(tensor_list toutputs, list_of_retainable & outputs) {
-  for(auto & o : toutputs) {
-    outputs.push_back(toRetainableSteal(std::move(o)));
+template<typename T>
+T tensor_as(at::Tensor&& t) = delete;
+
+template<>
+int64_t tensor_as(at::Tensor&& t) {
```

Comment:
Can we put this in some common header? I will need this in my dispatch PR as well